Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -0,0 +1,332 @@
+    get() = when (kind) {
+        is PropertyStub.Kind.Val -> kind.getter.flags
+        is PropertyStub.Kind.Var -> kind.getter.flags
+        is PropertyStub.Kind.Constant -> flagsOf()
+    }
+
+private val PropertyAccessor.Getter.flags: Flags
+    get() = listOfNotNull(
+            Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+            Flag.IS_PUBLIC,
+            Flag.IS_FINAL,
+            Flag.PropertyAccessor.IS_EXTERNAL.takeIf { this is PropertyAccessor.Getter.ExternalGetter }
+    ).let { flagsOf(*it.toTypedArray()) }
+
+private val PropertyStub.setterFlags: Flags
+    get() = if (kind !is PropertyStub.Kind.Var) flagsOf()
+    else kind.setter.flags
+
+private val PropertyAccessor.Setter.flags: Flags
+    get() = listOfNotNull(
+            Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+            Flag.IS_PUBLIC,
+            Flag.IS_FINAL,
+            Flag.PropertyAccessor.IS_EXTERNAL.takeIf { this is PropertyAccessor.Setter.ExternalSetter }
+    ).let { flagsOf(*it.toTypedArray()) }
+
+private val StubType.flags: Flags
+    get() = listOfNotNull(
+            Flag.Type.IS_NULLABLE.takeIf { nullable }
+    ).let { flagsOf(*it.toTypedArray()) }
+
+private val TypealiasStub.flags: Flags
+    get() = listOfNotNull(
+            Flag.IS_PUBLIC
+    ).let { flagsOf(*it.toTypedArray()) }
+
+private val FunctionParameterStub.flags: Flags
+    get() = listOfNotNull(
+            Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() }
+    ).let { flagsOf(*it.toTypedArray()) }
+
+private fun AnnotationStub.map(): KmAnnotation {
+    val args = when (this) {
+        AnnotationStub.ObjC.ConsumesReceiver -> TODO()
+        AnnotationStub.ObjC.ReturnsRetained -> TODO()
+        is AnnotationStub.ObjC.Method -> TODO()
+        is AnnotationStub.ObjC.Factory -> TODO()
+        AnnotationStub.ObjC.Consumed -> TODO()
+        is AnnotationStub.ObjC.Constructor -> TODO()
+        is AnnotationStub.ObjC.ExternalClass -> TODO()
+        AnnotationStub.CCall.CString -> mapOf()
+        AnnotationStub.CCall.WCString -> mapOf()
+        is AnnotationStub.CCall.Symbol ->
+            mapOf("id" to KmAnnotationArgument.StringValue(symbolName))
+        is AnnotationStub.CStruct -> TODO()
+        is AnnotationStub.CNaturalStruct -> TODO()
+        is AnnotationStub.CLength -> TODO()
+        is AnnotationStub.Deprecated -> TODO()
+    }
+    return KmAnnotation(classifier.fqNameSerialized, args)
+}
+
+/**
+ * @param shouldExpandTypeAliases describes how should we write type aliases.
+ * If [shouldExpandTypeAliases] is true then type alias-based types are written as
+ * ```
+ * Type {
+ *  abbreviatedType = AbbreviatedType.abbreviatedClassifier
+ *  classifier = AbbreviatedType.underlyingType
+ *  arguments = AbbreviatedType.underlyingType.typeArguments
+ * }
+ * ```
+ * So we basically replacing type alias with underlying class.
+ * NB: We are assuming here that underlying class is already expanded.
```

Comment:
Is it?
`KotlinClassifierType.toStubIrType()` implementations looks like it can produce `AbbreviatedType(AbbreviatedType(...)...)`.